GetServiceByRevision/handler.ts

```diff
@@ -103,30 +105,30 @@ const getServiceByRevisionTask = (
         }
       ],
       query: `SELECT * FROM m WHERE m.${SERVICE_MODEL_ID_FIELD} = @serviceId and m.version = @version`
-    })
-
-    .chain(maybeService =>
-      taskEither.of(
-        isSome(maybeService)
+    }),
+    TE.chain(maybeService =>
+      TE.of(
+        O.isSome(maybeService)
           ? ResponseSuccessJson(retrievedServiceToPublic(maybeService.value))
           : ResponseErrorNotFound(
               "Service not found",
               "The service you requested was not found in the system."
             )
       )
-    );
+    )
```

Comment:
Here you can use `O.foldW` instead of `O.isSome`, in order to avoid using the `value` property which should be considered as an internal detail of the `fp-ts` library.

Using `flow`, you can also reduce the code nesting by one level

```suggestion
    TE.chain(flow(
        O.foldW(
            () => ResponseErrorNotFound("Service not found", "The service you requested was not found in the system."),
            (service) => ResponseSuccessJson(retrievedServiceToPublic(service)),
        ),
        TE.of,
    ))
```

Reply:
You're absolutely right, thank you. Done in [7c31690](https://github.com/pagopa/io-functions-assets/pull/43/commits/7c316905bcf976015f9166b0e720d9bfe39682ce)